src/tools/wasm-split.cpp

```diff
@@ -777,6 +820,77 @@ void splitModule(Module& wasm, const WasmSplitOptions& options) {
   writeModule(*secondary, options.secondaryOutput, options);
 }
 
+void mergeProfiles(const WasmSplitOptions& options) {
+  // Read the initial profile. We will merge other profiles into this one.
+  ProfileData data = readProfile(options.inputFiles[0]);
+
+  // In verbose mode, we want to find profiles that don't contributed to the
+  // merged profile. To do that, keep track of how many profiles each function
+  // appears in. If any profile contains only functions that appear in multiple
+  // profiles, it could be dropped.
```

Comment:
But why do we want to drop it and/or report it is redundant? That's not really an error IIUC... maybe just one profile happened to be a "sub-profile" of another.

Reply:
This is a feature request from https://github.com/WebAssembly/binaryen/issues/3912#issuecomment-850189858. This information is not an error, but it is useful for paring down the number of workloads used to create the combined profile. (Of course it would be more useful if we implemented a solution to the [set cover problem](https://en.wikipedia.org/wiki/Set_cover_problem), but this seems good enough for now.)

Reply:
I am a little worried of feature creep here. But if it's requested then I'm not strongly opposed.

Reply:
I agree. It may make more sense to refactor this into a separate mode or flag in the future rather than using `verbose` as a catch-all flag for dumping information. I also think it will make sense to split wasm-split up into multiple files soon, which should help with the maintenance burden of all these random features.